street_network/src/lib.rs

```diff
@@ -3,6 +3,7 @@ extern crate anyhow;
 #[macro_use]
 extern crate log;
 
+use std::cell::RefCell;
```

Comment:
https://doc.rust-lang.org/std/cell/index.html is a bit of a Rust trick here. In the middle of a transformation, we might be immutably borrowing `StreetNetwork`, because we're iterating over all roads or something. If we want to write some debug info and record it in the `StreetNetwork`, how can we do that? `RefCell` lets us cheat and do something mutable with an immutable reference. It just does a double-borrow check at runtime. It's always safe & simple in this case; we have some methods to push a labelled point and immediately relinquish the borrow.

The alternative would be passing around some kind of mutable `Debugger` object to the transformations and accumulating debug state there. That might be nicer to switch to eventually, but for the moment, the API would be more awkward -- we'd probably have to do something like `debugger.label_intersection(&streets, "something here")`